### ISSUE TITLE
Side Bar drag to hide feature

### ISSUE THREAD

**Issue by TejasGhanwat** [marktext/marktext#1601]
It will be great to have the side bar hiding feature by dragging it to left instead of going to "View" option and selecting"Toggle Sidebar" option. Similar to the feature that Visual Studio Code offers.

**Comment by fxha** [comment]
I think it's sufficient to add a right click menu to hide it. If you just want to hide the sidebar without icons, you can click on the selected panel icon again.

> Similar to the feature that Visual Studio Code offers.

Could you please explain what vscode offers?

**Comment by AngelGzS** [comment]
@fxha I think he refers to this feature in VS Code

![vscode](https://user-images.githubusercontent.com/177025/68326878-df9ba980-0091-11ea-8cac-2ae9c9e6a2b8.gif)

**Comment by Lisiadito** [comment]
I took a look at this and it's super little effort to implement this.
![Peek 2019-11-07 23-08](https://user-images.githubusercontent.com/13214912/68431921-a3f90080-01b3-11ea-8e98-027ba761e59c.gif)

One thing I would like to fix additionally is that the width of the sidebar contains and grows with the word count.

**Comment by Lisiadito** [comment]
Ah I just realized that it belongs to the titlebar and not the side bar... any ideas?

**Comment by Jocs** [comment]
@TejasGhanwat Do you mean completely hiding the sideBar? Is it the same as the toggle sidebar hiding function in the menu?

**Comment by Jocs** [comment]
>Ah I just realized that the word count belongs to the titlebar and not the side bar... any ideas?

I prefer hide all the sidebar, so it will not have this problem?

**Comment by Lisiadito** [comment]
Ok lets wait what feature @TejasGhanwat actually requested. But problem with the word count being displayed not ideally is already an existing problem we should maybe open a issue for. Because its possible to toggle the right column by pressing the symbols as well.

![Peek 2019-11-08 10-19](https://user-images.githubusercontent.com/13214912/68464844-a1c58f00-0211-11ea-8998-ebc952abd3e9.gif)

**Comment by fxha** [comment]
> But problem with the word count being displayed not ideally is already an existing problem we should maybe open a issue for. Because its possible to toggle the right column by pressing the symbols as well.

Exactly, there was another issue with (very) long file paths that may overlap too but I closed it as expected behavior. @Lisiadito Please feel free to open an issue to discuss about changes because we don't really know how to fix this. We could change the themes styled but that would look not great.

**Comment by Jocs** [comment]
Thank you for the suggestion! While drag-to-hide has not been implemented, MarkText already offers two quick ways to collapse the sidebar:

1. **Click the active panel icon** in the left icon strip — this collapses the right panel, leaving just the narrow icon column visible (identical to the VS Code icon-click behavior shown in the comments).
2. **View → Toggle Sidebar** (or the keyboard shortcut) hides the sidebar entirely.

Given these existing shortcuts cover the core use case and the request has been open without a PR for over six years, we are going to close this as a stale feature request. If there is renewed interest in a drag-threshold approach, please open a new issue with a concrete proposal and we will be happy to revisit it.
